lib/rules/require-await.js

```diff
@@ -42,14 +42,47 @@ module.exports = {
         schema: [],
 
         messages: {
-            missingAwait: "{{name}} has no 'await' expression."
-        }
+            missingAwait: "{{name}} has no 'await' expression.",
+            removeAsync: "Remove 'async'."
+        },
+
+        hasSuggestions: true
     },
 
     create(context) {
         const sourceCode = context.sourceCode;
         let scopeInfo = null;
 
+        /**
+         * Finds the range that covers the `async` keyword.
+         * @param {FunctionNode} node The function node to find the `async` keyword in.
+         * @returns {[number, number] | null} The range of the `async` keyword, or null if it could not be found.
+         */
+        function findAsyncKeywordRange(node) {
+
+            /*
+             * If the function belongs to a method definition or
+             * property, then the function's range will not include the
+             * `async` keyword. We need to look at the parent instead.
+             */
+            const nodeWithAsyncKeyword =
+                node.parent.type === "MethodDefinition" ||
+                node.parent.type === "Property"
+                    ? node.parent
+                    : node;
+
+            const text = context.sourceCode.getText(nodeWithAsyncKeyword);
+            const match = /(^|\b)async\b/u.exec(text);
+
+            if (match) {
+                const start = nodeWithAsyncKeyword.range[0] + match.index;
+
+                return [start, start + 5];
+            }
+
+            return null;
```

Comment:
```suggestion
            const asyncToken = sourceCode.getFirstToken(nodeWithAsyncKeyword, token => token.value === "async");

            return [asyncToken.range[0], sourceCode.getTokenAfter(asyncToken).range[0]]
```

We can simplify this by using tokens, and also remove extra whitespace before the next token.

Reply:
Oh, sweet! I didn't realise you could get the individual tokens. 🤦‍♂️